tests/test_e2e.py

```diff
@@ -58,56 +57,35 @@ def test_e2e_plan(cli_runner: CliRunner, data_dir: Path, rules_dir: Path) -> Non
     assert result.exit_code == 0
 
 
-@freeze_time("2023-05-12 12:12:53")
-@pytest.mark.timeout(5)
 def test_e2e_gui(
-    cli_runner: CliRunner,
     unused_tcp_port: int,
     data_dir: Path,
     tmp_path: Path,
-    thread_executor: ThreadPoolExecutor,
-    mocker: MockerFixture,
 ) -> None:
-    def client_select_directory(port: int) -> httpx.Response:
-        # It's probably overkill to start an event loop,
-        # but wait_for_port provides exactly what's needed here.
-        # Use a timeout so a failing test won't be held up by this thread.
-        asyncio.run(asyncio.wait_for(wait_for_port(port), timeout=2))
-        return httpx.post(
-            f"http://127.0.0.1:{port}/redact/",
-            data={
-                "input_directory": str(data_dir / "input" / "tiff"),
-                "output_directory": str(tmp_path),
-            },
-        )
-
-    webbrowser_open_mock = mocker.patch("webbrowser.open")
-    # Ideally, we'd be able to use a running event loop to schedule the client
-    # requests. However, the CLI starts its own event loop, then blocks until
-    # completion. We could hook into or patch internal application events to
-    # get a callback with an already-running event loop, but that provides less
-    # isolation than just running the client in an independent thread.
-    # Note, Click's CliRunner expects to run in a single-threaded environment,
-    # but the new thread won't use any of the patched stdout, etc. streams and
-    # a ProcessPoolExecutor is slower and requires more delicate pickling. so
-    # we'll break that expection here.
-    client_future = thread_executor.submit(client_select_directory, unused_tcp_port)
-
-    cli_result = cli_runner.invoke(main.imagedephi, ["gui", "--port", str(unused_tcp_port)])
-    # If an HTTP request is successfully sent, either the normal response from this endpoint or a
-    # 500 error should cause the CLI to halt. However, if halting fails, then this test is
-    # particularly susceptible to exceeding the timeout, which on Windows will crash pytest (due
-    # to the behavior of pytest-timeout).
-
-    assert cli_result.exit_code == 0
-    webbrowser_open_mock.assert_called_once()
-    output_file = tmp_path / "Redacted_2023-05-12_12-12-53" / "study_slide_1.tif"
+
+    port = unused_tcp_port
+    subprocess.Popen(
```

Comment:
What happens to this subprocess when this test case ends? Does it keep running in the background until the parent Pytest terminates?

Reply:
explicitly killed in c95f49ee9c4caa7ae50536b73b038fcb266dcc81